ticketbuyer/purchase.go

```diff
@@ -65,4 +74,25 @@ out:
 			break out
 		}
 	}
+	p.wg.Done()
+}
+
+// Start starts the purchase manager goroutines.
+func (p *PurchaseManager) Start() {
+	p.wg.Add(1)
+	go p.NotificationHandler()
+}
+
+// WaitForShutdown blocks until all purchase manager goroutines have finished executing.
+func (p *PurchaseManager) WaitForShutdown() {
+	p.wg.Wait()
+}
+
+// Stop signals all purchase manager goroutines to shutdown.
+func (p *PurchaseManager) Stop() {
+	select {
+	case <-p.quit:
+	default:
+		close(p.quit)
```

Comment:
It's not common to hit but this is not concurrent safe because the entire operation is not atomic.  Need to lock this with a mutex.